### PR TITLE
Subplot registration

### PR DESCRIPTION
**Pull request by mdtusz**
Some slight changes so that the `plot` modules will be loaded along with their respective traces. 

This will allow us to slim down the bundle size even more in the case where only a few traces (e.g. no gl) are needed.

### CODE REVIEW

File: src/plots/plots.js [hunk 1]
```diff
@@ -41,7 +41,7 @@ plots.fontWeight = 'normal';
  */
 plots.register = function(_module, thisType, categoriesIn, meta) {
     if(modules[thisType]) {
-        console.warn('type ' + thisType + ' already registered');
+        console.log('type ' + thisType + ' already registered');
```
Comment (etpinard) [inline src/plots/plots.js]: Bringing in @bpostlethwaite and @alexcjohnson 

Should we warn (or spit out a console.log) when users register the same plot module twice?

I'd vote for doing nothing at all.

@mdtusz If I'm no mistaken, every time a user registers two or more trace modules that rely on the same plot module e.g. 

```js
var PlotlyBar = require('plotly.js/lib/bar');
var PlotlyBox = require('plotly.js/lib/box');

Plotly.register([PlotlyBar, PlotlyBox]);
```

would split out a log.

File: src/plots/plots.js [hunk 2]
```diff
@@ -132,32 +132,12 @@ plots.registerSubplot = function(_module) {
...
-    var attr = _module.attr,
-        idRoot = _module.idRoot;
-
-    var regexStart = '^',
-        regexEnd = '([2-9]|[1-9][0-9]+)?$',
-        hasXY = (plotType === 'cartesian' || subplotsRegistry === 'gl2d');
-
-    function makeRegex(mid) {
-        return new RegExp(regexStart + mid + regexEnd);
+        console.log('plot type ' + plotType + ' already registered');
```
Comment (etpinard) [inline src/plots/plots.js]: Ooops the above comment (https://github.com/plotly/plotly.js/pull/193/files#r50156440) was supposed to be placed here.

Comment (mdtusz) [inline src/plots/plots.js]: It *would*, but there's a check in the `Plotly.register` function that will avoid calling `subplotsRegistry` if the module has already been loaded.

File: src/plotly.js [hunk 3]
```diff
@@ -80,6 +67,10 @@ exports.register = function register(_modules) {
             throw new Error('Invalid module was attempted to be registered!');
         } else {
             Plots.register(newModule, newModule.name, newModule.categories, newModule.meta);
+
+            if(!Plots.subplotsRegistry[newModule.basePlotModule.name]){
```
Comment (mdtusz) [inline src/plotly.js]: @etpinard right here, we make the check to see if the plot was registered.

Comment (etpinard) [inline src/plotly.js]: haha. Nice!

File: src/plots/plots.js (hunk 1 above)
Comment (mdtusz) [inline src/plots/plots.js]: Fyi I switched log --> warn out of (perhaps bad) habit, from when I had a project that needed to support ie8 :frowning:

Comment (etpinard) [inline src/plots/plots.js]: Oh boy. No way plotly.js works in ie8

**Comment by mdtusz** [comment]
How are we feeling on this one @etpinard?

**Comment by etpinard** [comment]
:dancer: let's merge this in.

We can revisit the log vs warn issue later.

**Pull request merged**